ledger/src/ui/test/attestation/test_attestation.c

```diff
@@ -0,0 +1,385 @@
+    assert(rawkey == (unsigned char *)G_att_ctx.priv_key_data);
+    assert(key == &G_att_ctx.priv_key);
+    assert(KEYLEN == key_len);
+    set_private_key(key, rawkey);
+    return 0;
+}
+
+int cx_ecfp_generate_pair(cx_curve_t curve,
+                          cx_ecfp_public_key_t *pubkey,
+                          cx_ecfp_private_key_t *privkey,
+                          int keepprivate) {
+    assert(CX_CURVE_256K1 == curve);
+    assert(pubkey == &G_att_ctx.pub_key);
+    assert(privkey == &G_att_ctx.priv_key);
+    ASSERT_MEMCMP(PRIVATE_KEY, privkey->d, KEYLEN);
+    assert(1 == keepprivate);
+    set_public_key(pubkey, PUBLIC_KEY);
+    return 0;
+}
+
+// os mocks
+unsigned int os_endorsement_get_code_hash(unsigned char *buffer) {
+    // Should not be called for the current implementation
+    ASSERT_FAIL();
+}
+
+unsigned int os_endorsement_key2_derive_sign_data(unsigned char *src,
+                                                  unsigned int srcLength,
+                                                  unsigned char *signature) {
+    assert(src == (unsigned char *)G_att_ctx.msg);
+    assert(srcLength == G_att_ctx.msg_offset);
+    assert(signature == APDU_DATA_PTR);
+    memcpy(signature, MSG_SIGNATURE, sizeof(MSG_SIGNATURE) - 1);
+
+    return sizeof(MSG_SIGNATURE) - 1;
+}
+
+void os_perso_derive_node_bip32(cx_curve_t curve,
+                                unsigned int *path,
+                                unsigned int pathLength,
+                                unsigned char *privateKey,
+                                unsigned char *chain) {
+    assert(CX_CURVE_256K1 == curve);
+    assert(path == (unsigned int *)G_att_ctx.path);
+    assert(privateKey == (unsigned char *)G_att_ctx.priv_key_data);
+    ASSERT_MEMCMP(path, PUBKEY_PATH, PUBKEY_PATH_LENGTH);
+    ASSERT_MEMCMP(privateKey, PRIVATE_KEY, KEYLEN);
+    assert(NULL == chain);
+}
+
+void os_memmove(void *dst, const void *src, unsigned int length) {
+    memmove(dst, src, length);
+}
+
+// signer_authorization mocks
+sigaut_signer_t *get_authorized_signer_info() {
+    memcpy(G_signer_info.hash, SIGNER_HASH, sizeof(SIGNER_HASH));
+    G_signer_info.iteration = 9;
+    return &G_signer_info;
+}
+
+// Unit tests
+void test_reset_attestation() {
+    printf("Test reset attestation...\n");
+    memcpy(G_att_ctx.msg, "a-msg", strlen("a-msg"));
+    G_att_ctx.msg_offset = strlen("a-msg");
+    memcpy(G_att_ctx.path, "a-path", strlen("a-path"));
+    set_private_key(&G_att_ctx.priv_key, (unsigned char *)PRIVATE_KEY);
+    set_public_key(&G_att_ctx.pub_key, PUBLIC_KEY);
+
+    reset_attestation(&G_att_ctx);
+    ASSERT_STRUCT_CLEARED(att_t, G_att_ctx);
+    assert(att_stage_wait_ud_value == G_att_ctx.stage);
+}
+
+void test_get_attestation_ud_value() {
+    printf("Test ATT_OP_UD_VALUE...\n");
+    unsigned int rx;
+
+    reset_attestation(&G_att_ctx);
+    *(unsigned char *)N_onboarded_ui = 1;
+    memcpy(G_att_ctx.priv_key_data, PRIVATE_KEY, sizeof(PRIVATE_KEY));
```

Comment:
This is part of the behaviour we are testing, so we shouldn't be doing it as part of the test setup. Sorry, I didn't spot it in the previous review. This should be done by the derivation function from the path.